src/cr/cube/dimension.py

```diff
@@ -19,124 +21,125 @@ class Dimension(object):
     :attr:`.CrunchCube.dimensions`.
     """
 
-    def __init__(self, dim, selections=None):
-        self._dim = dim
-        self._type = self._get_type(dim, selections)
+    def __init__(self, dimension_dict, next_dimension_dict=None):
+        self._dimension_dict = dimension_dict
+        self._next_dimension_dict = next_dimension_dict
 
     @lazyproperty
     def alias(self):
-        """Alias of a cube's dimension."""
-        refs = self._dim['references']
+        """str system (as opposed to human) name for this dimension."""
+        refs = self._dimension_dict['references']
         return refs.get('alias')
 
     @lazyproperty
     def description(self):
-        """Description of a cube's dimension."""
-        refs = self._dim['references']
-        return refs.get('description')
+        """str description of this dimension."""
+        description = self._dimension_dict['references'].get('description')
+        return description if description else ''
+
+    @lazyproperty
+    def dimension_type(self):
+        """str representing type of this cube dimension."""
+        # ---all this logic really belongs in the Dimensions collection
+        # ---object, which is where it will move to once that's implemented
+
+        def next_dim_is_mr_cat():
+            """True if subsequent dimension is an MR_CAT dimension."""
+            if not self._next_dimension_dict:
+                return False
+
+            categories = self._next_dimension_dict['type'].get('categories')
+            if not categories:
+                return False
```

Comment:
This line is untested. Is there an easy way of testing it?

Reply:
This whole method is reimplemented in the next PR (to come) that extracts dimension type discovery into the Dimensions collection. I propose we skip elaborating the test to exercise this line since it will only be cut out a commit or two later to be replaced by a more permanent test elsewhere.